src/assets/scripts/client/aircraft/AircraftController.js

```diff
@@ -111,52 +111,65 @@ export default class AircraftController {
      * This is the entry method for creating new departing and arriving aircraft.
      * This method should only be called as a callback from a `SpawnScheduler` timer.
      *
-     * @method createAircraftWithSpawnModel
+     * @method createAircraftWithSpawnPatternModel
      * @param spawnPatternModel {SpawnPatternModel}
+     * @private
      */
-    createAircraftWithSpawnModel = (spawnPatternModel) => {
+    createAircraftWithSpawnPatternModel = (spawnPatternModel) => {
         const initializationProps = this._buildAircraftProps(spawnPatternModel);
-        const aircraftModel = new AircraftInstanceModel(initializationProps);
 
-        this.addItem(aircraftModel);
+        this._createAircraftWithInitializationProps(initializationProps);
     }
 
     /**
-     * @for AircraftController
-     * @method aircraft_auto_toggle
+     * Build aircraft props for `spawnPatternModel` and add
+     * preSpawn data to `baseAircraftProps`.
+     *
+     * Used when creating aircraft already along an arrival route on load
+     * or on airport change.
+     *
+     * This method should be called direectly and not via a timer callback
+     *
+     * This hooks into the same method used to build spawnning aircraft
+     * and simply adds another layer on top of that to build a preSpawn aircraft.
+     *
+     * @method createPreSpawnAircraftWithSpawnPatternModel
+     * @param  spawnPatternModel {SpawnPatternModel}
+     * @private
      */
-    aircraft_auto_toggle() {
-        this.aircraft.auto.enabled = !this.aircraft.auto.enabled;
-    }
+    createPreSpawnAircraftWithSpawnPatternModel = (spawnPatternModel) => {
+        for (let i = 0; i < spawnPatternModel.preSpawnAircraftList.length; i++) {
+            const preSpawnHeadingAndPosition = spawnPatternModel.preSpawnAircraftList[i];
+            const baseAircraftProps = this._buildAircraftProps(spawnPatternModel);
+            const initializationProps = Object.assign({}, baseAircraftProps, preSpawnHeadingAndPosition);
```

Comment:
ha!  You are correct I would poke you but it would be for the reasoning and not necessarily the use.  Using `Object.assign` is totally cool within an es2015 pipeline.  its like using `.slice(0)` on an array.  

Because I would ask you the same, here is my reasoning:  we get our preSpawn specific props from `preSpawnHeadingAndPosition`.  we get the usual aircraft props from `baseAircraftProps`, like any other aircraft.  Then, we _create a new object_ from those two objects, with the `preSpawnHeadingAndPosition` object taking precedence over `baseAircraftProps`.  So if both objects contain a `position` key, the position from `preSpawnHeadingAndPosition` will be used.
example:
```javascript
const base = {
    altitude: 0,
    speed: 0,
    flightNumber: '123',
    airline: 'aal'
};

const preSpawn = {
    altitude: 10000,
    speed: 240
}

const initProps = Object.assign({}, base, preSpawn);

// we would end up with
initProps: {
    airline: 'aal',
    altitude: 10000,
    flightNumber: '123',
    speed: 240
}
```
This means we can use our existing propsBuilder method to build aircraft props as usual, then just add the preSpawn props over the top. 💥